node/node_state.rei

```diff
@@ -25,6 +25,31 @@ type t = {
   validators_uri: Address_map.t(Uri.t),
   recent_operation_results:
     BLAKE2B.Map.t([ | `Transaction | `Withdraw(Ledger.Handle.t)]),
+  /** A list of triples including:
+    - State root hash - an indentifier for a state root onto which you can apply
+          blocks and know the current state
+    - Validator hash - the hash of all the validators approved in corresponding
+          state root.
+    - Epoch of the state root hash - a timestampe generated when the hash
+          of the state root began to be created.
+
+    FIXME: clarify terminology of epoch vs finality period. Basically same thing.
+    (Actually I've started to clarify this more in flows.re [try_to_produce_block])
+
+    The state root starts with genesis, and then updates every finality
+    period (approximately every 60 seconds). This means you don't have to
+    download the entire history of the chain to derive the state corresponding
+    to a given block - only the state corresponding to its state root hash is required.
+
+    At the start of each new finality period, we start hashing the current
+    state, requiring every validator to finish hashing that same state before
+    the start of the next finality period. At the end of the finality
+    period, the state hash finished in that period becomes the hash used
+    in the next finality period. Because a block always points to the state
+    hash of the previous period, nodes does not need to wait for the expensive
+    operation of hashing to complete before they can apply a block, and can thus
+    hash the state asynchronously. */
+  state_root_hash_list: list((BLAKE2B.t, BLAKE2B.t, float)),
```

Comment:
Make this a record, the comment is a nice to have, but it would be a lot more clear if it was on the type definition of the what a `state_root_hash_list` means here.

Also I believe this can be it's own module as it models the domain of the system.

Reply:
Also I don't understand why this contains the list of validators hash, what does it achieve?

Reply:
I didn't understand an offline comment you made about validator hashes. Fixed here: https://github.com/marigold-dev/deku/pull/195/commits/08de202b8d9b9ea950c8cd2a29ef3a108bcbc4e5